tests/integration-tests/tests/schedulers/test_custom_munge_key.py

```diff
@@ -0,0 +1,194 @@
+        custom_munge_key_arn=custom_munge_key_arn,
+    )
+    cluster.update(str(update_cluster_stop_login_config))
+
+    # wait for LoginNodes gracetime_period
+    check_login_nodes_stopped(remote_command_executor)
+
+    # Test rotation script runs successfully
+    result = remote_command_executor.run_remote_command("sudo /opt/parallelcluster/scripts/slurm/update_munge_key.sh")
+    exit_code = result.return_code
+    assert_that(exit_code).is_equal_to(0)
+
+    update_cluster_remove_custom_munge_key_config = pcluster_config_reader(
+        config_file="pcluster.remove_custom_munge_key.config.yaml"
+    )
+    cluster.update(str(update_cluster_remove_custom_munge_key_config))
+
+    # Test Munge Key has been changed
+    _test_custom_munge_key_fetch_and_decode(
+        remote_command_executor, encode_custom_munge_key, use_custom_munge_key=False
+    )
+
+    # Test if munge key is fully functional after cluster roll back
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "fail-on-node-updated.sh"), "fail-on-node-updated.sh")
+    update_cluster_fail_roll_back_config = pcluster_config_reader(
+        config_file="pcluster.roll_back.config.yaml",
+        custom_munge_key_arn=custom_munge_key_arn,
+        bucket_name=bucket_name,
+    )
+    cluster.update(str(update_cluster_fail_roll_back_config), raise_on_error=False)
+
+    cluster.start()
+    wait_for_computefleet_changed(cluster, "RUNNING")
+    scheduler_commands.submit_command_and_assert_job_accepted(
+        submit_command_args={
+            "command": "srun sleep 1",
+            "nodes": 2,
+        }
+    )
+
+
+def generate_secure_random_key(length=64):
+    return os.urandom(length)
+
+
+def create_base64_encoded_key():
+    random_key = generate_secure_random_key()
+    base64_encoded_key = base64.b64encode(random_key).decode("utf-8")
+
+    return base64_encoded_key
+
+
+@retry(wait_fixed=seconds(20), stop_max_delay=minutes(15))
+def check_login_nodes_stopped(remote_command_executor):
+    result = remote_command_executor.run_remote_command(
+        "sudo /opt/parallelcluster/scripts/slurm/check_login_nodes_stopped.sh",
+        raise_on_error=False,
+    )
+    exit_code = result.return_code
+    assert_that(exit_code).is_equal_to(0)
+
+
+def _test_custom_munge_key_fetch_and_decode(
+    remote_command_executor, encode_custom_munge_key, use_custom_munge_key=True
+):
+    """Test encoded munge key in secret manager has been successfully fetched by cluster and decode."""
+    result = remote_command_executor.run_remote_command("sudo cat /etc/munge/munge.key | base64")
+    encode_munge_key = result.stdout.strip().replace("\n", "")
+    if use_custom_munge_key:
+        assert_that(encode_munge_key).is_equal_to(encode_custom_munge_key)
+    else:
+        assert_that(encode_munge_key).is_not_equal_to(encode_custom_munge_key)
```

Comment:
I would propose a minor refactoring here:
- since this function just checks what munge key is currently in use, let's rename it to `check_encoded_munge_key()`;
- change the variable name `encode_custom_munge_key` to `expected_encoded_munge_key`;
- change the variable name `encode_munge_key` to `encoded_munge_key`;

Reply:
Also, there's a tiny chance that the last assertion will fail, because the probability of the `mungekey` utility to generate a munge key equal to our custom munge key is non-zero (although *very* small).

But we like to live dangerously. Let's keep it.

Reply:
Done!